src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -971,6 +971,7 @@ def create_python_udf_or_sp(
     secure: bool = False,
     external_access_integrations: Optional[List[str]] = None,
     secrets: Optional[Dict[str, str]] = None,
+    immutable: bool = False,
```

Comment:
from the definition I think this immutable is optional: https://docs.snowflake.com/en/sql-reference/sql/create-function#python-handler

so I think the parameters should be `immutable: Optional[bool] = None`
```suggestion
    immutable: bool = False,
```

Reply:
oh you're right, I overlooked the part that it has a default value, thanks for calling it out.
now I'm thinking whether the signature should be `immutable = False` or `volatile = True`

@sfc-gh-jdu @sfc-gh-sfan @sfc-gh-aalam do you know if this parameter is supported in the scala udf?

Reply:
Any luck on this PR review?

For naming I'll advocate for immutable since the antonym is mutable, but volatile doesn't have that.
TBH, I'll prefer deterministic as a better naming, but that's inconsistent with SQL keyword, so let's keep it.

Reply:
I'm okay with naming it `immutable=False`. Other option is we have something like `mutability : Literal["volatile", "immutable"] = "volatile"`

Reply:
@sfc-gh-aling shall we proceed with this?

Reply:
yes, I'm also okay with immutable=False